src/woo_publications/publications/tests/test_documents_admin.py

```diff
@@ -218,3 +220,119 @@ def test_publications_admin_delete(self):
+            form["document_uuid"] = uuid.uuid4()
+            form["identifier"] = uuid.uuid4()
+            form["officiele_titel"] = "The official title of this document"
+            form["creatiedatum"] = "2024-01-01"
+
+            add_response = form.submit(name="_save")
+
+            self.assertRedirects(
+                add_response, reverse("admin:publications_document_changelist")
+            )
+            added_item = Document.objects.order_by("-pk").first()
+            assert added_item is not None
+
+        with self.subTest(
+            "provided no service and uuid configured creates item with no errors"
+        ):
+            form = response.forms["document_form"]
+            form["publicatie"] = publication.id
+            form["document_service"].select(text="---------")
+            form["document_uuid"] = ""
+            form["identifier"] = uuid.uuid4()
+            form["officiele_titel"] = "The official title of this document"
+            form["creatiedatum"] = "2024-01-01"
+
+            add_response = form.submit(name="_save")
+
+            self.assertRedirects(
+                add_response, reverse("admin:publications_document_changelist")
+            )
+            added_item = Document.objects.order_by("-pk").first()
+            assert added_item is not None
+
+        with self.subTest(
+            "provided only service and no uuid configured results in error"
+        ):
+            form = response.forms["document_form"]
+            form["publicatie"] = publication.id
+            form["document_service"].select(text=str(service))
+            form["document_uuid"] = ""
+            form["identifier"] = uuid.uuid4()
+            form["officiele_titel"] = "The official title of this document"
+            form["creatiedatum"] = "2024-01-01"
+
+            add_response = form.submit(name="_save")
+
+            self.assertEqual(add_response.status_code, 200)
+            self.assertContains(
+                add_response,
+                "You must specify both the Document API Service "
+                "and Document UUID to identify a document.",
+            )
+
+        with self.subTest(
+            "provided only uuid and no service configured results in error"
+        ):
+            form = response.forms["document_form"]
+            form["publicatie"] = publication.id
+            form["document_service"].select(text="---------")
+            form["document_uuid"] = uuid.uuid4()
+            form["identifier"] = uuid.uuid4()
+            form["officiele_titel"] = "The official title of this document"
+            form["creatiedatum"] = "2024-01-01"
+
+            add_response = form.submit(name="_save")
+
+            self.assertEqual(add_response.status_code, 200)
+            self.assertContains(
+                add_response,
+                "You must specify both the Document API Service "
+                "and Document UUID to identify a document.",
+            )
```

Comment:
I don't think these tests are necessary - I trust that the Django admin properly displays the violation error message when the constraint is being violated. No need to test the framework itself.

The model layers test is sufficient.